File: test/ruby/DataTypeMapIntInt.rb
#!/usr/bin/env ruby
require 'hyperdex'

def assert &block
    raise RuntimeError unless yield
end

c = HyperDex::Client::Client.new(ARGV[0], ARGV[1].to_i)
assert { c.put("kv", "k", {}) == true }
assert { c.get("kv", "k") == {:v => {}} }
assert { c.put("kv", "k", {"v" => {1 => 7, 2 => 8, 3 => 9}}) == true }
assert { c.get("kv", "k") == {:v => {1 => 7, 2 => 8, 3 => 9}} }
assert { c.put("kv", "k", {"v" => {}}) == true }
assert { c.get("kv", "k") == {:v => {}} }

File: test/python/DataTypeMapIntInt.py
#!/usr/bin/env python2
import sys
import hyperdex.client
from testlib import * 

from hyperdex.client import LessEqual, GreaterEqual, LessThan, GreaterThan, Range, Regex, LengthEquals, LengthLessEqual, LengthGreaterEqual
c = hyperdex.client.Client(sys.argv[1], int(sys.argv[2]))
def to_objectset(xs):
    return set([frozenset(x.items()) for x in xs])
    
assertTrue(c.put('kv', 'k', {}))
assertEquals(c.get('kv', 'k'), {'v': {}})
assertTrue(c.put('kv', 'k', {'v': {1: 7, 2: 8, 3: 9}}))
assertEquals(c.get('kv', 'k'), {'v': {1: 7, 2: 8, 3: 9}})
assertTrue(c.put('kv', 'k', {'v': {}}))
assertEquals(c.get('kv', 'k'), {'v': {}})
assertTrue(c.put('kv', 'k', {'v': {1: 7, 2: 8, 3: 9}}))

# Atomic Operations
assertTrue(c.map_atomic_add('kv', 'k', {'v': {1: 5}}))
assertTrue(c.put('kv', 'k', {'v': {1: 11, 2: 8, 3: 9}}))
assertEquals(c.get('kv', 'k'), {'v': {1: 7, 2: 8, 3: 9}})
assertTrue(c.map_atomic_min('kv', 'k', {'v': {3: 100}}))
assertEquals(c.get('kv', 'k'), {'v': {1: 7, 2: 8, 3: 9}})
assertTrue(c.map_atomic_min('kv', 'k', {'v': {3: 1}}))
assertEquals(c.get('kv', 'k'), {'v': {1: 7, 2: 8, 3: 1}})
assertTrue(c.map_atomic_max('kv', 'k', {'v': {3: 100}}))
assertEquals(c.get('kv', 'k'), {'v': {1: 7, 2: 8, 3: 100}})
